pkg/storage/unified/resource/cdk_blob.go

```diff
@@ -0,0 +1,185 @@
+package resource
+
+import (
+	"bytes"
+	context "context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"mime"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/noop"
+	"gocloud.dev/blob"
+
+	// Supported drivers
+	_ "gocloud.dev/blob/azureblob"
+	_ "gocloud.dev/blob/fileblob"
+	_ "gocloud.dev/blob/gcsblob"
+	_ "gocloud.dev/blob/memblob"
+	_ "gocloud.dev/blob/s3blob"
```

Comment:
Most of the additions to `go.mod` are because this is ready to support any of these engines -- however they are not loaded directly, only though the URL configs

Reply:
AFAIK `memblob` and `fileblob` are for testing and would not be used in production, could they be imported only in `_test.go` files, or otherwise in separate directories dedicated to testing? We can also build a testing binary using go build tags while avoiding the testing dependencies for production binaries entirely:
```go
//go:build testing
package main

import (
	_ "gocloud.dev/blob/memblob"
	_ "gocloud.dev/blob/fileblob"
)

// This file can live in grafana/pkg/cmd/grafana/testing.go
// There is nothing else to do here. It will just import the
// testing libraries when we add the `testing` go build tag.
// Example just running the service locally:
//
//  go run -race -tags testing ./pkg/cmd/grafana -- server target
//
// Example first building, then running:
//
//  go build -race -tags testing -o bin/grafana-testing ./pkg/cmd/grafana
//  ./bin/grafana-testing server target
```

Importing testing code into production code increases the attack surface and the possibility of configuration errors.